source/constructs/api/data_source/service.py

```diff
@@ -2581,29 +2673,356 @@ def query_connection_detail(account: JDBCInstanceSourceBase):
+    return f'report_{time_str}'
+
+def __check_empty_for_field(row, header):
+    if row[0].value is None or str(row[0].value).strip() == const.EMPTY_STR:
+        return True, f"{header[0]} should not be empty"
+    if row[1].value is None or str(row[1].value).strip() == const.EMPTY_STR:
+        return True, f"{header[1]} should not be empty"
+    if row[3].value is None or str(row[3].value).strip() == const.EMPTY_STR:
+        return True, f"{header[3]} should not be empty"
+    if row[5].value is None or str(row[5].value).strip() == const.EMPTY_STR:
+        if row[6].value is None or str(row[6].value).strip() == const.EMPTY_STR:
+            return True, f"{header[6]} should not be empty when {header[5]} is empty"
+        if row[7].value is None or str(row[7].value).strip() == const.EMPTY_STR:
+            return True, f"{header[7]} should not be empty when {header[5]} is empty"
+    if row[8].value is None or str(row[8].value).strip() == const.EMPTY_STR:
+        return True, f"{header[8]} should not be empty"
+    if row[9].value is None or str(row[9].value).strip() == const.EMPTY_STR:
+        return True, f"{header[9]} should not be empty"
+    if row[10].value is None or str(row[10].value).strip() == const.EMPTY_STR:
+        return True, f"{header[10]} should not be empty"
+    return False, None
+
+def __map_network_jdbc(created_jdbc_list, subnets, security_group_id):
+    res = []
+    for index, item in enumerate(created_jdbc_list):
+        item.network_sg_id = security_group_id
+        item.network_subnet_id = subnets[0] if index % 2 == 0 else subnets[1]
+        res.append(item)
+    return res
 
-def list_jdbc_databases(source: JdbcSource) -> list[str]:
-    url_arr = source.connection_url.split(":")
-    if len(url_arr) != 4:
-        raise BizException(MessageEnum.SOURCE_JDBC_URL_FORMAT_ERROR.get_code(), MessageEnum.SOURCE_JDBC_URL_FORMAT_ERROR.get_msg())
-    if url_arr[1] != "mysql":
-        raise BizException(MessageEnum.SOURCE_JDBC_LIST_DATABASES_NOT_SUPPORTED.get_code(), MessageEnum.SOURCE_JDBC_LIST_DATABASES_NOT_SUPPORTED.get_msg())
-    host = url_arr[2][2:]
-    port = int(url_arr[3].split("/")[0])
-    user = source.username
-    password = source.password
-    if source.secret_id:
-        secrets_client = boto3.client('secretsmanager')
-        secret_response = secrets_client.get_secret_value(SecretId=source.secret_id)
-        secrets = json.loads(secret_response['SecretString'])
-        user = secrets['username']
-        password = secrets['password']
-    mysql_database = jdbc_database.MySQLDatabase(host, port, user, password)
-    databases = mysql_database.list_databases()
-    logger.info(databases)
-    return databases
+def query_batch_status(filename: str):
+    success, warning, failed = 0, 0, 0
+    file_key = f"{const.BATCH_CREATE_REPORT_PATH}/{filename}.xlsx"
+    response = __s3_client.list_objects_v2(Bucket=admin_bucket_name, Prefix=const.BATCH_CREATE_REPORT_PATH)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `list_objects_v2` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `ContinuationToken`, `IsTruncated`, `NextContinuationToken`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)